x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/tree/Node.java

```diff
@@ -85,6 +86,12 @@ public void forEachUp(Consumer<? super T> action) {
         action.accept((T) this);
     }
 
+    @SuppressWarnings("unchecked")
+    public void forEachUp(Consumer<? super T> action, Function<? super T, Collection<T>> childrenGetter) {
```

Comment:
I'm not sure if this should be local to where I need it, or if it's fine to have this here. Not sure if it will be reused or not

Reply:
Better to move it where you needed it. You can choose to do two separate iterations or a generic one where you filter the nodes based on their type to decide the action.